app/submissions/views.py

```diff
@@ -1,9 +1,24 @@
 from rest_framework import viewsets
-
+from rest_framework import generics
 from .models import Submission
 from .serializers import SubmissionSerializer
+from django.http import HttpResponse
+from wsgiref.util import FileWrapper
+from django.shortcuts import get_object_or_404
 
 
 class SubmissionViewList(viewsets.ModelViewSet):
     serializer_class = SubmissionSerializer
     queryset = Submission.objects.all()
+
+
+class FileDownloadListAPIView(generics.ListAPIView):
```

Comment:
Why inheriting from ListAPIView? It is used to retrieve a collection of instances. RetrieveAPIView is a better choice here.